Zend/zend_API.c

```diff
@@ -3723,58 +3724,151 @@ ZEND_API int zend_declare_property_stringl(zend_class_entry *ce, const char *nam
 }
 /* }}} */
 
-ZEND_API int zend_declare_class_constant(zend_class_entry *ce, const char *name, size_t name_length, zval *value) /* {{{ */
+ZEND_API int zend_declare_class_constant_ex(zend_class_entry *ce, zend_string *name, zval *value, int access_type, zend_string *doc_comment) /* {{{ */
```

Comment:
The RFC says we are mirroring method visibilities in interfaces. As far as I see, this method does not prohibit private and protected visibilities for class constants… I guess the check was just missed:
```C
	if (ce->ce_flags & ZEND_ACC_INTERFACE) {
		if ((op_array->fn_flags & ZEND_ACC_PPP_MASK) != ZEND_ACC_PUBLIC) {
			zend_error_noreturn(E_COMPILE_ERROR, "Access type for interface constant %s::%s must be omitted", ZSTR_VAL(ce->name), ZSTR_VAL(name));
		}
	}
```